simulator/event_manager.go

```diff
@@ -179,7 +185,7 @@ type EventHistoryCollector struct {
 
 	m    *EventManager
 	page *ring.Ring
-	pos  int
+	pos  *list.Element
```

Comment:
Checked all reads/writes on this field and they seem to be protected by holding a lock on the manager.